public/src/main/java/com/revenuecat/purchases/models/PurchaseDetails.kt

```diff
@@ -0,0 +1,39 @@
+package com.revenuecat.purchases.models
+
+import android.os.Parcelable
+import com.android.billingclient.api.Purchase
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.parceler.JSONObjectParceler
+import kotlinx.android.parcel.Parcelize
+import kotlinx.android.parcel.TypeParceler
+import org.json.JSONObject
+
+@Parcelize
+@TypeParceler<JSONObject, JSONObjectParceler>()
+// TODO: docs
+data class PurchaseDetails(
+    val orderId: String,
+    val sku: String,
+    val type: ProductType,
+    val purchaseTime: Long,
+    val purchaseToken: String,
+    val purchaseState: RevenueCatPurchaseState,
+
+    /*
+     * Null for restored purchases
+     */
+    val isAutoRenewing: Boolean?,
+
+    /*
+     * Null for Amazon
+     */
+    val signature: String?,
+
+    val originalJson: JSONObject
+) : Parcelable
+
+// TODO: should this be nullable or just throw
+val PurchaseDetails.originalGooglePurchase: Purchase?
```

Comment:
I am going to be thinking this in that other PR where I move PurchaseWrapper to be a `PurchaseDetails`